fix: get correct context directly before showing dialog

RyanCarrier opened the pull request:
Correct context as late as possible before displaying dialog to make sure it doesn't change before use, also check it's mounted 

Issue most likely rises from the Future.delayed in checkVersion

fixes #323

larryaasen commented on lib/src/upgrade_alert.dart:
```diff
@@ -117,11 +117,7 @@ class UpgradeAlertState extends State<UpgradeAlert> {
-                  ? widget.navigatorKey!.currentContext!
-                  : context;
-              checkVersion(context: checkContext);
+              checkVersion(context: context);
```
@RyanCarrier By not using `widget.navigatorKey!.currentContext` in the function `checkVersion`, it is not used by downstream functions, such as `widget.upgrader.determineMessages`, which eventually is used to determine the locale. I don't fully understand the implications of this change.

larryaasen commented on lib/src/upgrade_alert.dart:
```diff
@@ -230,10 +226,19 @@ class UpgradeAlertState extends State<UpgradeAlert> {
     showDialog(
       barrierDismissible: barrierDismissible,
-      context: context,
+      context: checkContext,
```
@RyanCarrier I generally agree with this change. Good find.

RyanCarrier commented:
Thanks for this, I added a test and pretty sure I've figured out what's happening.
In my case I have a a few things that happen before checking for updates which seems to be causing a rebuild, but I hold my Upgrader object. So it seems that initialize is getting called on the Upgrader, then the UpgraderAlert is getting given a new context. 
So I added a test that calls initialize first, which fails without the fix. The fix appears to still work even with the context 'corrected' before the checkVersion call so shouldn't have any Locale concerns :)

RyanCarrier commented:
Hey mate I assume this is closed due to moving to main. Are you happy for me to re-open and resync for main?
